File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    const answer = arr1.map((el,i)=>{
        const temp = el | arr2[i];
        return temp.toString(2).padStart(n,"0").replace(/1/g,"#").replace(/0/g," ");
    })
    return answer;
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        int[] map = new int[n];
        
        for(int i=0; i<n; i++){
            map[i] = arr1[i] | arr2[i];
        }
        
        for(int i=0; i<n; i++){
            String s = Integer.toString(map[i], 2);
            System.out.println(s);
            s = s.replace("1","#");
            s = s.replace("0", " ");
            if(s.length()<n){
                s = " ".repeat(n-(s.length()))+s;
            }
            answer[i] = s;
        }
        return answer;
    }
}